toolchain/lower/mangler.cpp

```diff
@@ -0,0 +1,99 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/lower/mangler.h"
+
+#include "toolchain/base/kind_switch.h"
+#include "toolchain/sem_ir/entry_point.h"
+
+namespace Carbon::Lower {
+
+auto Mangler::MangleInverseQualifiedNameScope(llvm::raw_ostream& os,
+                                              SemIR::NameScopeId name_scope_id)
+    -> void {
+  struct NameEntry {
+    SemIR::NameScopeId name_id;
+    char prefix;
+  };
+  llvm::SmallVector<NameEntry> names_to_render;
+  auto add_scope = [&](SemIR::NameScopeId name_scope_id, char prefix) {
+    if (name_scope_id.is_valid() &&
+        name_scope_id != SemIR::NameScopeId::Package) {
+      names_to_render.push_back({name_scope_id, prefix});
```

Comment:
Please use designated initializers for structs. We've gone in that direction for [struct style](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/cpp_style_guide.md) because it matches what we'd do in Carbon, and avoids some initialization nuances of C++.

Note, I'd also be agnostic if you switch to a `pair` instead of a struct, since we call out an exception for that as tuple-ish.

Reply:
Sure thing, done (kept with a named struct and added designated initializers).